### DISCUSSION TITLE
Git Blame vs. Full History Parsing

### DISCUSSION THREAD

**Discussion by mircealungu** [git-truck/git-truck#90] [General]
Folks, 

One thing just occurred to me.

Why do we need to go back in history to look at contributions.

When we can just use git blame on a given version in order to get our corresponding contribution percentages? 

Why is this better for ownership? Because no matter how much I worked two year ago on File A, if Thomas has rewritten all of it and from my old code there's only a single line left... then, for all practical purposes he knows the file more than me. 

Now, this is not to see that the backwards analyzing parser is not useful. We can still use it, and we SHOULD use it, in order to compute e.g. churn for a given file... that is, how much has it changed. Because files that change a lot are very likely most important in a system. 

Not to mention that, while the user is looking at the GitBlame based vis we can parse the history and make the other vis available later ;)

**Comment by hojelse** [comment]
This is probably a point of discussion to inform the design of user interface (view options ect.)

You could maybe call Git Blame and Full History Parsing, collectively as scopes? And Heatmap, coldmap, file type and truckfactor for metrics?

Scopes:
- Full History scope (code churn)
- Current version scope (survived lines, blame)

Metrics:
- Heatmap metric
- Coldmap metric
- File type metric
- Truckfactor metric
